example/tests/errors_test.py

```diff
@@ -0,0 +1,15 @@
+import origen
+import pytest
+from origen.errors import *
+
+def boot_falcon():
+    if origen.app is None:
+        return origen.app.instantiate_dut("dut.falcon")
+    else:
+        return origen.app
+
+def test_duplicate_detected_error():
+    app = boot_falcon()
+    with pytest.raises(DuplicateDetectedError) as err_info:
+        raise DuplicateDetectedError(app)
+    assert str(err_info.value) == "Cannot create instance of class 'example.application.Application' named 'example', it already exists!"

```

Comment:
I think we should go back to calling this your original name (DuplicateInstance ?).
The reason I said the more generic name was that I imagined that it could be used for many different duplicate scenarios and the error message would be defined by the specific case.
However, here the error message couples it to a duplicate instance (which is nice that this module takes care of it), so we should call it a more descriptive error name as you originally had it.

Reply:
I made the pytest a real world example, let me know if my `sub_blocks` implementation looks correct.